### PR TITLE
Added changes supporting renaming of habitat-api to habitat-lab

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
We changed the name of `Habitat-API` to `Habitat Lab` to make less confusing and stress the purpose of the repo to propose Tasks, Metrics, Baselines to the community. Mirroring https://github.com/facebookresearch/habitat-lab/pull/448 for Habitat Lab.

- Fixed the referenced links that become broken on the website
- Added `--depth 1` option to speed up Hab Lab clone step

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -281,7 +281,7 @@ We highly recommend installing a [miniconda](https://docs.conda.io/en/latest/min
   `640 x 480, total time: 3.208 sec. FPS: 311.7`.
   Note that the test scenes do not provide semantic meshes.
   If you would like to test the semantic sensors via `example.py`, please use the data from the Matterport3D dataset (see [Datasets](#Datasets)).
-  We have also provided an [example demo](https://aihabitat.org/docs/habitat-api/habitat-api-demo.html) for reference.
+  We have also provided an [example demo](https://aihabitat.org/docs/habitat-lab/habitat-lab-demo.html) for reference.
```
Comment (Skylion007) [inline README.md:284]: make sure to actually rename these

File: .circleci/config.yml [hunk 2]
```diff
@@ -402,11 +402,11 @@ jobs:
           command: |
               export PATH=$HOME/miniconda/bin:$PATH
               . activate habitat;
-              if [ ! -d ./habitat-api ]
+              if [ ! -d ./habitat-lab ]
               then
-                git clone -q https://github.com/facebookresearch/habitat-api.git
+                git clone -q https://github.com/facebookresearch/habitat-lab.git
```
Comment (Skylion007) [inline .circleci/config.yml:407]: nit: we should add --depth 1 so it runs faster (aka does a shallow clone)

Comment (mathfac) [reply .circleci/config.yml:407]: HabLab doesn't have deps submodules.

**Comment by mathfac** [comment]
Checking CI passes and documentation build with https://github.com/facebookresearch/habitat-lab/pull/448 merged.

File: .circleci/config.yml (hunk 2 above)
Comment (Skylion007) [reply .circleci/config.yml:407]: That's now what it means by shallow. Shallow means it only checkouts the most recent files, not the history or older commits.

**Pull request merged**